multistream-select/src/protocol/dialer.rs

```diff
@@ -149,7 +150,32 @@ impl<R> Stream for Dialer<R>
 
 			} else {
 				// A varint number of protocols
-				unimplemented!()
+				let mut reader = Cursor::new(frame);
+				let num_protocols: usize = varint::decode(reader.by_ref())?;
```

Comment:
Should this be synchronous? What if the underlying stream blocks?

Reply:
When we reach this point, the entire data has already arrived. `frame` is a `Bytes`, not a `impl AsyncRead`.